test/lib/server/services/run/RunService.test.js

```diff
@@ -310,4 +310,137 @@ module.exports = () => {
             { runPatch: { runQuality: RunQualities.TEST } },
         );
     });
+
+    it('should successfully update run with eorReasons with reasonTypeId', async () => {
+        const runNumber = 1;
+
+        const eorReasons = [
+            {
+                reasonTypeId: 1,
+                description: 'test1',
+            },
+            {
+                reasonTypeId: 1,
+                description: 'test2',
+            },
+        ];
+
+        const run = await runService.update(
+            { runNumber },
+            { relations: { eorReasons: eorReasons } },
+        );
+
+        expect(run.eorReasons).to.lengthOf(2);
+        // Description is unique, should be equal
+        for (let i = 0; i < run.eorReasons.length; i++) {
+            expect(run.eorReasons[i].description).to.equal(eorReasons[i].description);
+        }
+    });
+
+    it('should successfully update run with eorReasons with category and title', async () => {
+        const runNumber = 1;
+
+        const eorReasons = [
+            {
+                category: 'DETECTORS',
+                title: 'CPV',
+                description: 'test1',
+            },
+            {
+                category: 'DETECTORS',
+                title: 'TPC',
+                description: 'test2',
+            },
+        ];
+
+        const run = await runService.update(
+            { runNumber },
+            { relations: { eorReasons: eorReasons } },
+        );
+
+        expect(run.eorReasons).to.lengthOf(2);
+        // Description is unique, should be equal
+        for (let i = 0; i < run.eorReasons.length; i++) {
+            expect(run.eorReasons[i].description).to.equal(eorReasons[i].description);
```

Comment:
Same thing here, you should be able to test the title, category and even that the id it the one expected (check in the seeders which reasonTypeId correspond to this title & category)

Reply:
The title, category and id could be checked here, but the seeders are more hardcoded right, so is it not a bad idea to check for hardcoded seeders information for the reasonTypeId?

Reply:
Thank you for the explanation, I have implemented this now in the tests as you requested.